CustomerAPI/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Data.Entities;
+using Data.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CustomerAPI.Controllers
+{
+    [Route("api/controller")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private ICustomerService _service;
+
+        public CustomerController(ICustomerService service)
+        {
+            _service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        [HttpPost]
+        public IActionResult Create(Customer customer)
+        {
+            try
+            {
+                _service.Create(customer);
+                return Created("Id: ", customer.Id);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        [HttpGet("{Id}")]
+        public IActionResult GetById(long Id)
+        {
+            try
+            {
+                var customer = _service.GetById(Id);
+                return Ok(customer);
+            }
+            catch (ArgumentNullException exception)
+            {
+                return NotFound(exception.Message);
+            }
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            try
+            {
+                var listCustomers = _service.GetAll();
+                return Ok(listCustomers);
+            }
+            catch (Exception exception)
+            {
+                var exceptionMessage = exception.InnerException?.Message ?? exception.Message;
+                return Problem(exceptionMessage);
+            }
+        }
+
+        [HttpPut]
+        public IActionResult Update(Customer customer)
+        {
+            try
+            {
+                _service.Update(customer);
+                return Ok();
+            }
+            catch (ArgumentNullException exception)
+            {
+                return NotFound(exception.Message);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        [HttpDelete]
+        public IActionResult Delete(long Id)
+        {
+            try
+            {
+                _service.Delete(Id);
+                return NoContent();
+            }
+            catch (ArgumentException exception)
+            {
+                return NotFound(exception.Message);
+            }
```

Comment:
Quando tratares de qualquer exceção, é interessante sempre colocar também um catch mais genérico, para pegar qualquer outro problema que não estava previsto para acontecer. Isso vale para os outros métodos dessa classe também. (Problem é o erro 500)
O ideal seria:
```suggestion
            }
            catch(Exception exception)
            {
               return Problem(exception.Message);
            }
```